envoy/filesystem/filesystem.h

```diff
@@ -169,15 +169,19 @@ struct DirectoryEntry {
   // target. For example, if name_ is a symlink to a directory, its file type will be Directory.
   FileType type_;
 
+  // The file size in bytes for regular files. Should not be relied on for directories,
+  // symlinks, or FileType::Other.
+  uint64_t size_bytes_;
```

Comment:
Any way we can make this less of a footgun? One thing that comes to mind is wrapping it in a call that ASSERTs that `type_` is `Regular`? Or perhaps making it a absl::optional?

Reply:
Asserting that `type_` is `Regular` doesn't actually solve the problem, because on Windows with a symlink the type is regular and the size is unknown (or maybe it's the size of the linked path, I forget - either way, not the size of the file).

Using `absl::optional` is a good idea though, to make it a hard failure instead of a quiet failure if size is used when it shouldn't be.